File: go-aws-whoami/whoami.go
package main

import (
	"context"
	"fmt"
	"log"

	_ "github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/service/sts"
)

func main() {
	fmt.Println("+++ AWS whoami +++")

	cfg, err := config.LoadDefaultConfig(context.TODO())
	if err != nil {
		log.Fatal(err)
	}

	client := sts.NewFromConfig(cfg)

	input := sts.GetCallerIdentityInput{}
	output, err := client.GetCallerIdentity(context.TODO(), &input)

	if err != nil {
		log.Fatal(err)
	}

	fmt.Printf("User ID: %s\n", *output.UserId)
	fmt.Printf("Account: %s\n", *output.Account)
	fmt.Printf("ARN: %s\n", *output.Arn)
}

File: node-aws-whoami-v2/whoami.js

const AWS = require("aws-sdk");
const chain = new AWS.CredentialProviderChain();

function getCredsType(creds) {
    const types = [
		"EnvironmentCredentials",
		"SsoCredentials",
		"SharedIniFileCredentials",
		"ECSCredentials",
		"ProcessCredentials",
		"TokenFileWebIdentityCredentials",
		"EC2MetadataCredentials",
    ];

    for (const type of types) {
        if (creds instanceof AWS[type]) {
            return type;
        }
    }
    return "Unknown";
}

chain.resolve((err, creds) => {
    console.log("+++ AWS whoami +++");

	if (err) {
		console.error("Failed to get credentials:", err);
        process.exit(1);
	}
	console.log(`Credentials type: ${getCredsType(creds)}`);

    const sts = new AWS.STS();
    sts.getCallerIdentity({}, (err, response) => {
        if (err) {
            console.error("Failed to get identity:", err);
			process.exit(1);
        }
		console.log(`User ID: ${response.UserId}`);
		console.log(`Account: ${response.Account}`);
		console.log(`ARN: ${response.Arn}`);
    });
});
